src/geometry/euclidean/basic.lean

```diff
@@ -843,6 +843,26 @@ begin
   exact (dec_trivial : (1 : fin 3) ≠ 2) (hfi hf12)
 end
 
+/-- Any three points in a cospherical set are affinely independent. -/
+lemma cospherical.affine_independent_of_mem_of_ne {s : set P} (hs : cospherical s) {p₁ p₂ p₃ : P}
+  (h₁ : p₁ ∈ s) (h₂ : p₂ ∈ s) (h₃ : p₃ ∈ s) (h₁₂ : p₁ ≠ p₂) (h₁₃ : p₁ ≠ p₃) (h₂₃ : p₂ ≠ p₃) :
+  affine_independent ℝ ![p₁, p₂, p₃] :=
+begin
+  refine hs.affine_independent _ _,
+  { simp [h₁, h₂, h₃, set.insert_subset] },
+  { intros i j hij,
+    fin_cases i; fin_cases j;
+      simpa [h₁₂, h₁₂.symm, h₁₃, h₁₃.symm, h₂₃, h₂₃.symm] using hij }
```

Comment:
I would guess that #17779 lets this be golfed even further.

Reply:
Changed to use `cons_injective_iff`; not sure why that requires `erw` rather than being usable directly with `simp`.